### PR TITLE
prefer `sys.platform` over `os.name` when checking for windows OS

### PR DESCRIPTION
**Pull request by m-rsha**

### CODE REVIEW

File: pre_commit/languages/conda.py [hunk 1]
```diff
@@ -26,7 +27,7 @@ def get_env_patch(env: str) -> PatchesT:
     # $CONDA_PREFIX/Scripts and $CONDA_PREFIX. Whereas the latter only
     # seems to be used for python.exe.
     path: SubstitutionT = (os.path.join(env, 'bin'), os.pathsep, Var('PATH'))
-    if os.name == 'nt':  # pragma: no cover (platform specific)
+    if sys.platform == 'win32':  # pragma: no cover (platform specific)
```
Comment (asottile) [inline pre_commit/languages/conda.py:30]: this can be `# pragma: win32 cover`

File: pre_commit/languages/python.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def norm_version(version: str) -> str | None:
     elif _sys_executable_matches(version):  # virtualenv defaults to our exe
         return None
 
-    if os.name == 'nt':  # pragma: no cover (windows)
+    if sys.platform == 'win32':  # pragma: no cover (windows)
```
Comment (asottile) [inline pre_commit/languages/python.py:140]: this can be `# pragma: win32 cover`

Comment (m-rsha) [reply pre_commit/languages/python.py:140]: It borked the tests, so I changed it back :D. I guess this conversation still counts as "resolved"?

**Review by asottile — APPROVED** [review]

**Pull request merged**